src/analysis/multimodal_util.py

```diff
@@ -808,6 +834,69 @@ def plot_license_terms_stacked_bar_chart_collections(
 
     return chart_licenses
 
+def plot_tasks_chart(
+    df, task_typemap, order, pwidth, pheight, save_dir, modality, tasks_column, category_mapping_dict
+):
+    df = df[df['Modality'] == modality]
+    df_sources = categorize_tasks(df, order, task_typemap, tasks_column, modality, category_mapping_dict)
+    df_sources = df_sources[df_sources[tasks_column].notnull()]
+
+    # Aggregate counts for each category
+    df_aggregated = df_sources[tasks_column].value_counts().reset_index()
+    df_aggregated.columns = [tasks_column, 'count']
+
+    # Calculate percentages and format with a '%' sign
+    df_aggregated['percentage'] = ((df_aggregated['count'] / df_aggregated['count'].sum()) * 100).round().astype(int)
+
+    sorted_order = df_aggregated.sort_values('count', ascending=False)[tasks_column].tolist()
+
+    bar_chart = alt.Chart(df_aggregated).mark_bar(size=30).encode(
+        y=alt.Y(
+            field=f"{tasks_column}",
+            type="nominal",
+            title=f"{tasks_column}",
+            sort=sorted_order,
+            axis=alt.Axis(labelFontSize=12, titleFontSize=14)
```

Comment:
pass in font sizes as args in the function so they only need to be changed in one place